### ISSUE TITLE
Execute test with fixed parameter and varargs result in IllegalArgumentException (All test data are parsed as array to all parameter)

### ISSUE THREAD

**Issue by aquadrehz** [cbeust/testng#920]
Execute test with fixed parameter and varargs result in IllegalArgumentException (All test data are parsed as array to all parameter)
I've create test method with fixed amount of parameter varargs. and execute as following
 - Java application (via Eclipse)
 - TestNG (via Eclipse plug-in)

The TestNG execution found IllegalArgumentException.
I found all test data are parsed as array to all parameter.

I suggest it result should be the same so this should be fixed.

```java
package test;

import org.testng.annotations.DataProvider;
import org.testng.annotations.Test;

public class testClass001 {
	public static void main(String[] args) {
		test("001", "002", "003", "004"); 
		// Execute as Java application.
		// The Length is 1
		// The Length is 3
	}
	
	@DataProvider(name = "testData")
    public static Object[][] testDataProvider() {
        return new Object[][] {new String[]{"001","002"},new String[]{"001","002","003", "004", "005"}};
		// Execute as Java application.
		// FAILED: test("[Ljava.lang.String;@7946e1f4", [Ljava.lang.String;@7946e1f4) java.lang.IllegalArgumentException: argument type mismatch
		// FAILED: test("[Ljava.lang.String;@3c09711b", [Ljava.lang.String;@3c09711b) java.lang.IllegalArgumentException: argument type mismatch
    }

    @Test(dataProvider = "testData")
    public static void test(String str001, String... str002) {
    	System.out.println("The Length is 1");
        System.out.println("The Length is "+str002.length);

    }
}



```

**Comment by nitinverma** [comment]
https://github.com/cbeust/testng/pull/923

TestClass for this issue: https://github.com/nvexperiments/testng-play/blob/master/issue-122/src/test/java/edu/nitin/testng/issue920/TestClass001.java

---

Running test case with 6.9.10

---
```
./build.sh -PtestngVersion=6.9.10
:clean
:compileJava
:processResources UP-TO-DATE
:classes
:jar
:startScripts
:distTar
:distZip
:assemble
:compileTestJava
:processTestResources
:testClasses
:test

Gradle Test Executor 1 STANDARD_OUT
    ...
    ... TestNG 6.9.10 by Cédric Beust (cedric@beust.com)
    ...


TestNG > Injection > edu.nitin.testng.issue112.CasesToFix.badTestIssue122[0]([Ljava.lang.Object;@6411ef83, [Ljava.lang.Object;@6411ef83) FAILED
    java.lang.IllegalArgumentException

TestNG > Injection > edu.nitin.testng.issue112.CasesToFix.badTestIssue122[1]([Ljava.lang.Object;@23e466a2, [Ljava.lang.Object;@23e466a2) FAILED
    java.lang.IllegalArgumentException

TestNG > Injection > edu.nitin.testng.issue112.CasesToFix.badTestIssue122[2]([Ljava.lang.String;@1f7cff03, [Ljava.lang.String;@1f7cff03) FAILED
    java.lang.IllegalArgumentException

TestNG > Injection > edu.nitin.testng.issue112.CasesToFix.goodTestIssue122[0](3, [Ljava.lang.String;@3dc4067e) STANDARD_OUT
    An item is "three"
    An item is "four"

TestNG > Injection > edu.nitin.testng.issue112.CasesToFix.goodTestIssue122[1](3, [Ljava.lang.String;@27894310) STANDARD_OUT
    An item is "three"
    An item is "four"

TestNG > Injection > edu.nitin.testng.issue112.CasesToFix.goodTestIssue122 FAILED
    org.testng.TestNGException

TestNG > Injection > edu.nitin.testng.issue112.CasesToFix.mixedArgs[0]([Ljava.lang.Object;@5c8d50cb, [Ljava.lang.Object;@5c8d50cb, [Ljava.lang.Object;@5c8d50cb, [Ljava.lang.Object;@5c8d50cb) FAILED
    java.lang.IllegalArgumentException

TestNG > Injection > edu.nitin.testng.issue112.CasesToFix.mixedArgs[1]([Ljava.lang.Object;@66baff54, [Ljava.lang.Object;@66baff54, [Ljava.lang.Object;@66baff54, [Ljava.lang.Object;@66baff54) FAILED
    java.lang.IllegalArgumentException

TestNG > Injection > edu.nitin.testng.issue920.TestClass001.test[0]([Ljava.lang.String;@1ca4f44d, [Ljava.lang.String;@1ca4f44d) FAILED
    java.lang.IllegalArgumentException

TestNG > Injection > edu.nitin.testng.issue920.TestClass001.test[1]([Ljava.lang.String;@25e70d01, [Ljava.lang.String;@25e70d01) FAILED
    java.lang.IllegalArgumentException

TestNG > Injection > edu.nitin.testng.issue920.TestClass001.test2[0]([Ljava.lang.String;@1ca4f44d, [Ljava.lang.String;@1ca4f44d) STANDARD_OUT
    The Object [Ljava.lang.String;@1ca4f44d/class [Ljava.lang.String;
    The Length is 2

TestNG > Injection > edu.nitin.testng.issue920.TestClass001.test2[1]([Ljava.lang.String;@25e70d01, [Ljava.lang.String;@25e70d01) STANDARD_OUT
    The Object [Ljava.lang.String;@25e70d01/class [Ljava.lang.String;
    The Length is 5

TestNG > Injection > edu.nitin.testng.issue920.TestClass001.test2[2]([Ljava.lang.Object;@2ff2e22a, [Ljava.lang.Object;@2ff2e22a) FAILED
    java.lang.IllegalArgumentException

TestNG > Injection > edu.nitin.testng.issue920.TestClass001.test2[3]([Ljava.lang.Object;@782f5985, [Ljava.lang.Object;@782f5985) FAILED
    java.lang.IllegalArgumentException

TestNG > Injection > edu.nitin.testng.issue920.TestClass001.test3[0]([Ljava.lang.Object;@2ff2e22a, [Ljava.lang.Object;@2ff2e22a) FAILED
    java.lang.IllegalArgumentException

TestNG > Injection > edu.nitin.testng.issue920.TestClass001.test4[0]([Ljava.lang.Object;@782f5985, [Ljava.lang.Object;@782f5985) FAILED
    java.lang.IllegalArgumentException

16 tests completed, 12 failed
:test FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///Users/nitin.verma/Codes/git/github/nvexperiments/testng-play/issue-122/build/reports/tests/index.html

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 1.659 secs
```


**After this fix**

---
```
:clean
:compileJava
:processResources UP-TO-DATE
:classes
:jar
:startScripts
:distTar
:distZip
:assemble
:compileTestJava
:processTestResources
:testClasses
:test

Gradle Test Executor 1 STANDARD_OUT
    ...
    ... TestNG 6.9.11-SNAPSHOT by Cédric Beust (cedric@beust.com)
    ...


TestNG > Injection > edu.nitin.testng.issue112.CasesToFix.badTestIssue122[0](3, [Ljava.lang.String;@a96f41) STANDARD_OUT
    An item is "three"
    An item is "four"

TestNG > Injection > edu.nitin.testng.issue112.CasesToFix.badTestIssue122[1](3, [Ljava.lang.String;@5e6f5732) STANDARD_OUT
    An item is "three"
    An item is "four"

TestNG > Injection > edu.nitin.testng.issue112.CasesToFix.badTestIssue122[2](3, [Ljava.lang.String;@64b27dd9) STANDARD_OUT
    An item is "three"
    An item is "four"

TestNG > Injection > edu.nitin.testng.issue112.CasesToFix.goodTestIssue122[0](3, [Ljava.lang.String;@a96f41) STANDARD_OUT
    An item is "three"
    An item is "four"

TestNG > Injection > edu.nitin.testng.issue112.CasesToFix.goodTestIssue122[1](3, [Ljava.lang.String;@5e6f5732) STANDARD_OUT
    An item is "three"
    An item is "four"

TestNG > Injection > edu.nitin.testng.issue112.CasesToFix.goodTestIssue122[2](3, [Ljava.lang.String;@c39115c) STANDARD_OUT
    An item is "three"
    An item is "four"

TestNG > Injection > edu.nitin.testng.issue112.CasesToFix.mixedArgs[0](3, true, [Ljava.lang.String;@3857d099, [Ljava.lang.String;@541318a6) STANDARD_OUT
    An item is "four"

TestNG > Injection > edu.nitin.testng.issue112.CasesToFix.mixedArgs[1](3, true, [Ljava.lang.String;@ea6d2c1, [Ljava.lang.String;@8f31894) STANDARD_OUT
    An item is "four"

TestNG > Injection > edu.nitin.testng.issue920.TestClass001.test[0](001, [Ljava.lang.String;@7f673649) STANDARD_OUT
    The String 001
    The Length is 1

TestNG > Injection > edu.nitin.testng.issue920.TestClass001.test[1](001, [Ljava.lang.String;@53db6ada) STANDARD_OUT
    The String 001
    The Length is 4

TestNG > Injection > edu.nitin.testng.issue920.TestClass001.test2[0](001, [Ljava.lang.String;@37e5ca17) STANDARD_OUT
    The Object 001/class java.lang.String
    The Length is 1

TestNG > Injection > edu.nitin.testng.issue920.TestClass001.test2[1](001, [Ljava.lang.String;@6f92128c) STANDARD_OUT
    The Object 001/class java.lang.String
    The Length is 4

TestNG > Injection > edu.nitin.testng.issue920.TestClass001.test2[2](920, [Ljava.lang.String;@34f2dca1) STANDARD_OUT
    The Object 920/class java.lang.Integer
    The Length is 4

TestNG > Injection > edu.nitin.testng.issue920.TestClass001.test2[3]([Ljava.lang.String;@150a3548, [Ljava.lang.String;@487a6cf0) STANDARD_OUT
    The Object [Ljava.lang.String;@150a3548/class [Ljava.lang.String;
    The Length is 4

TestNG > Injection > edu.nitin.testng.issue920.TestClass001.test3[0](920, [Ljava.lang.String;@5ecf754f) STANDARD_OUT
    The int 920
    The Length is 4

TestNG > Injection > edu.nitin.testng.issue920.TestClass001.test4[0]([Ljava.lang.String;@150a3548, [Ljava.lang.String;@15502937) STANDARD_OUT
    The array [0011, 0012]
    The Length is 4
:check
:build

BUILD SUCCESSFUL

Total time: 1.69 secs
```
---

**Comment by juherr** [comment]
Fixed by #923
